kura/org.eclipse.kura.db.sqlite.provider/src/main/java/org/eclipse/kura/internal/db/sqlite/provider/SqliteMessageStoreImpl.java

```diff
@@ -0,0 +1,452 @@
+        this.sqlAllUnpublishedMessages = SELECT_MESSAGE_METADATA_FROM + this.sanitizedTableName
+                + " WHERE publishedOn IS NULL ORDER BY priority ASC, createdOn ASC;";
+        this.sqlAllInFlightMessages = SELECT_MESSAGE_METADATA_FROM + this.sanitizedTableName
+                + " WHERE publishedOn IS NOT NULL AND qos > 0 AND confirmedOn IS NULL AND droppedOn IS NULL "
+                + "ORDER BY priority ASC, createdOn ASC";
+        this.sqlAllDroppedInFlightMessages = SELECT_MESSAGE_METADATA_FROM + this.sanitizedTableName
+                + " WHERE droppedOn IS NOT NULL ORDER BY priority ASC, createdOn ASC;";
+        this.sqlUnpublishAllInFlightMessages = UPDATE + this.sanitizedTableName
+                + " SET publishedOn = NULL WHERE publishedOn IS NOT NULL AND qos > 0 AND confirmedOn IS NULL;";
+        this.sqlDropAllInFlightMessages = UPDATE + this.sanitizedTableName
+                + " SET droppedOn = ? WHERE publishedOn IS NOT NULL AND qos > 0 AND confirmedOn IS NULL;";
+        this.sqlDeleteDroppedMessages = DELETE_FROM + this.sanitizedTableName
+                + " WHERE droppedOn <= ? AND droppedOn IS NOT NULL;";
+        this.sqlDeleteConfirmedMessages = DELETE_FROM + this.sanitizedTableName
+                + " WHERE confirmedOn <= ? AND confirmedOn IS NOT NULL;";
+        this.sqlDeletePublishedMessages = DELETE_FROM + this.sanitizedTableName
+                + " WHERE qos = 0 AND publishedOn <= ? AND publishedOn IS NOT NULL;";
+        this.sqlDeleteMessage = DELETE_FROM + this.sanitizedTableName + " WHERE id = ?;";
+
+        execute(this.sqlCreateTable);
+        execute(this.sqlCreateIndex);
+
+        createIndex(sanitizeSql(this.tableName + "_PUBLISHEDON"), this.sanitizedTableName, "(PUBLISHEDON DESC)");
+        createIndex(sanitizeSql(this.tableName + "_CONFIRMEDON"), this.sanitizedTableName, "(CONFIRMEDON DESC)");
+        createIndex(sanitizeSql(this.tableName + "_DROPPEDON"), this.sanitizedTableName, "(DROPPEDON DESC)");
+    }
+
+    private String sanitizeSql(final String string) {
+        final String sanitizedName = string.replace("\"", "\"\"");
+        return "\"" + sanitizedName + "\"";
+    }
+
+    // ----------------------------------------------------------
+    //
+    // Start/Stop, ServiceId
+    //
+    // ----------------------------------------------------------
+
+    // ----------------------------------------------------------
+    //
+    // Message APIs
+    //
+    // ----------------------------------------------------------
+
+    @Override
+    public synchronized int getMessageCount() throws KuraStoreException {
+
+        return withConnection(c -> {
+            try (final PreparedStatement stmt = c.prepareStatement(this.sqlMessageCount);
+                    final ResultSet rs = stmt.executeQuery()) {
+                if (rs.next()) {
+                    return rs.getInt(1);
+                } else {
+                    return -1;
```

Comment:
Should we return 0 here? You will get a -1 even if the table exists but has no rows.

Reply:
I agree that -1 is not ok here, I have moved the code to a utility class and have it throw an exception instead since it seems that H2 and SQLite always return a result set for this query.